packages/core/src/components/fields/MultiSchemaField.js

```diff
@@ -132,7 +132,10 @@ class AnyOfField extends Component {
     }
 
     const enumOptions = options.map((option, index) => ({
```

Comment:
Does it make sense to instead just set `option = retrieveSchema(option, registry.rootSchema)` earlier in this function, so that everything related to the `option` is already dereferenced (this might fix other bugs with oneOf / refs)?

Reply:
Sure. I guess I'll update the samples as well then since this will ignore any title that is a sibling to a $ref. 

Actually; if you look the reference sample. There you have a reference to the same schema but with different titles. That wouldn't be possible if we changed like above. We could implement your suggestion but still look for title in the $ref schema. 

What do you think?

Reply:
I've moved the logic to `SchemaField` instead, so props.options are now dereferenced. That also fixes #1819